From 37c2df0f19401a3004747dfb8604e6457a9a5302 Mon Sep 17 00:00:00 2001
Message-Id: <37c2df0f19401a3004747dfb8604e6457a9a5302.1299776671.git.jdenemar@redhat.com>
From: Osier Yang <jyang@redhat.com>
Date: Thu, 10 Mar 2011 15:02:28 +0800
Subject: [PATCH] storage: Update qemu-img flag checking

For newer qemu-img, the help string for "backing file format" is
"[-F backing_fmt]".

Fix the wrong logic error by commit e997c268.

* src/storage/storage_backend.c
(cherry picked from commit d99937695435f96af401c91b2e7f25e7868d7bce)

BZ#: 676984
https://bugzilla.redhat.com/show_bug.cgi?id=676984
---
 src/storage/storage_backend.c |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/storage/storage_backend.c b/src/storage/storage_backend.c
index 4f0096c..3466912 100644
--- a/src/storage/storage_backend.c
+++ b/src/storage/storage_backend.c
@@ -598,7 +598,8 @@ static int virStorageBackendQEMUImgBackingFormat(const char *qemuimg)
 
     start = strstr(help, " create ");
     end = strstr(start, "\n");
-    if ((tmp = strstr(start, "-F fmt")) && tmp < end)
+    if (((tmp = strstr(start, "-F fmt")) && tmp < end) ||
+        ((tmp = strstr(start, "-F backing_fmt")) && tmp < end))
         ret = QEMU_IMG_BACKING_FORMAT_FLAG;
     else if ((tmp = strstr(start, "[-o options]")) && tmp < end)
         ret = QEMU_IMG_BACKING_FORMAT_OPTIONS;
@@ -760,7 +761,7 @@ virStorageBackendCreateQemuImg(virConnectPtr conn,
         };
 
         if (vol->target.encryption != NULL) {
-            if (imgformat == QEMU_IMG_BACKING_FORMAT_FLAG) {
+            if (imgformat == QEMU_IMG_BACKING_FORMAT_OPTIONS) {
                 imgargv[8] = "-o";
                 imgargv[9] = "encryption=on";
             } else {
@@ -840,7 +841,7 @@ virStorageBackendCreateQemuImg(virConnectPtr conn,
         };
 
         if (vol->target.encryption != NULL) {
-            if (imgformat == QEMU_IMG_BACKING_FORMAT_FLAG) {
+            if (imgformat == QEMU_IMG_BACKING_FORMAT_OPTIONS) {
                 imgargv[6] = "-o";
                 imgargv[7] = "encryption=on";
             } else {
-- 
1.7.4.1

